spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/procedures/RewriteDataFilesProcedure.java

```diff
@@ -52,6 +52,7 @@ class RewriteDataFilesProcedure extends BaseProcedure {
           ProcedureParameter.required("table", DataTypes.StringType),
           ProcedureParameter.optional("strategy", DataTypes.StringType),
           ProcedureParameter.optional("sort_order", DataTypes.StringType),
+          ProcedureParameter.optional("z_order", DataTypes.StringType),
```

Comment:
Is it violating the stored procedure's contract to add a new argument in the middle?

Reply:
I initially kept it in the end. But then I thought it is good to be together and I "assumed" customers will use named arguments. 

I will move it to the end. Thanks.